### ISSUE TITLE
To identify available network interfaces more accurately

### ISSUE THREAD

**Issue by dikim33** [oscar-cluster/oscar#453]
**Reported by valleegr on 1 Feb 2008 18:40 UTC**
Submitted by Mengjuei Hsieh:

Hi,

I was playing around the oscar 5.1b1 scripts last night and found that
the network interface names that
/opt/oscar/scripts/system-sanity.d/network-check.pl detects are not
very accurate. For example, it could list 'table' and 'Iface' as
available interface in Fedora 8. This issue normally does not cause
any problem.

It turns out it's using 'netstat -nr' output for processing, but in
the trunk it uses ifconfig. So it should be more accurate in the
future. Nevertheless, for 5.1b1, we can apply this patch to fix the
problem.

```
--- system-sanity.d/network-check.pl.orig       2008-01-26 12:20:18.000000000 -0800
+++ system-sanity.d/network-check.pl    2008-01-28 09:23:19.125186013 -0800
@@ -83,7 +83,7 @@
 sub check_oscar_interface {
     my $oscar_if = $ENV{OSCAR_HEAD_INTERNAL_INTERFACE};
     my %nics;
-    open IN, "netstat -nr | awk \'{print \$NF}\' |" || die "ERROR: Unable to query NICs\n";
+    open IN, "netstat -nr | awk \'/\\./{print \$NF}\' | uniq |" || die "ERROR: Unable to query NICs\n";
     while( <IN> ) {
         chomp;
         next if /^\s/ || /^lo$/;
```

**Comment by dikim33** [comment]
**Modified by valleegr on 16 May 2008 00:38 UTC**

**Comment by dikim33** [comment]
**Comment by valleegr on 30 May 2008 09:24 UTC**
(In [7011]) Fixes #454: apply patch from Mengjuei Hsieh - detect network interface more 
accuratly.
